### PR TITLE
Added ability to set password strength configuration.

### PR DESCRIPTION
**Pull request by baleks**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `develop` branch, _not_ the `master` branch if no hotfix
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `Fix #xxx[,#xxx]`, where "xxx" is the Github issue number)
- [ ] All tests are passingp
- [x] New/updated tests are included
- [ ] Changelog was modified

If adding a **new feature**, the PR's description includes:
- [x] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

File: protected/humhub/tests/config/common.php [hunk 1]
```diff
@@ -4,8 +4,8 @@
     'components' => [
         'db' => [
             'dsn' => 'mysql:host=localhost;dbname=humhub_test',
-            'username' => 'travis',
-            'password' => '',
+            'username' => 'root',
```
Comment (danielkesselberg) [inline protected/humhub/tests/config/common.php]: Does this still works on travis?

File: protected/humhub/components/Application.php [hunk 2]
```diff
@@ -28,6 +28,11 @@ class Application extends \yii\web\Application
      */
     private $_homeUrl = null;
 
+    /**
+     * @var array additional password validation rules
+     */
+    public $passwordAdditionalRules = [];
```
Comment (danielkesselberg) [inline protected/humhub/components/Application.php]: Not sure if this is the right place to add this. Please review the documentation for config: http://docs.humhub.org/admin-advanced-configuration.html

Comment (baleks) [reply protected/humhub/components/Application.php]: This has been fixed, please review.

File: protected/config/common.php [hunk 3]
```diff
@@ -5,5 +5,4 @@
  * @see http://docs.humhub.org/admin-installation-configuration.html
  * @see http://docs.humhub.org/dev-environment.html
  */
-return [
-];
+return [];
```
Comment (buddh4) [inline protected/config/common.php]: Please don't commit this file, I normally use the assume unchanged git feature.

**Pull request merged**